mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -1180,34 +1181,44 @@ def search_prompts(
         Search for prompts in Unity Catalog.
 
         Args:
-            filter_string: Additional filter string (after catalog/schema are removed)
+            filter_string: Filter string. If catalog_name/schema_name not provided separately,
+                will parse them from filter_string expecting format:
+                "catalog = 'catalog_name' AND schema = 'schema_name'"
             max_results: Maximum number of results to return
             order_by: List of fields to order by (not used in current implementation)
             page_token: Token for pagination
             catalog_name: Unity Catalog catalog name (for UC registries)
             schema_name: Unity Catalog schema name (for UC registries)
         """
-        # Build the request with Unity Catalog schema if provided
-        if catalog_name and schema_name:
-            unity_catalog_schema = UnityCatalogSchema(
-                catalog_name=catalog_name, schema_name=schema_name
-            )
-            req_body = message_to_json(
-                SearchPromptsRequest(
-                    unity_catalog_schema=unity_catalog_schema,
-                    filter=filter_string,
-                    max_results=max_results,
-                    page_token=page_token,
+        # If catalog_name/schema_name not provided as separate args, parse from filter_string
+        if catalog_name is None or schema_name is None:
+            if filter_string:
+                catalog_name, schema_name, remaining_filter = (
+                    self._parse_catalog_schema_from_filter(filter_string)
                 )
-            )
-        else:
-            req_body = message_to_json(
-                SearchPromptsRequest(
-                    filter=filter_string,
-                    max_results=max_results,
-                    page_token=page_token,
+                filter_string = remaining_filter
```

Comment:
Our backend currently ignores remaining filter. @kriscon-db we should start throwing "unsupported" if filter is defined